geode-for-redis/src/main/java/org/apache/geode/redis/internal/services/RegionProvider.java

```diff
@@ -155,6 +160,18 @@ public String getRedisRegionName() {
     }
   }
 
+  @VisibleForTesting
+  static boolean areKeysCrossSlots(List<RedisKey> keysToLock) {
+    int slot = keysToLock.get(0).getSlot();
+    for (int i = 1, keysToLockSize = keysToLock.size(); i < keysToLockSize; i++) {
```

Comment:
I think this just iterates over all the elements in keysToLock with the exception of element 0. If this is correct, I think the code would be cleaner if you got a sublist from keysToLock and iterated over it. Declaring "keysToLockSize" in the "for" expression makes it look like something that might change in the loop but I think it never does.

Reply:
You're correct. I was originally iterating the whole list using a foreach loop, but since there wasn't any reason to compare the first key's slot to itself, I used IntelliJ's automatic refactor to turn it into an "optimized indexed for loop" which is what this is. I think the reason it's formatted like that with `keysToLockSize` defined in the for expression is so that we only call `keysToLock.size()` once to initialize it, rather than calling it every time the loop executes.

Given the cost of comparing the first key's slot to itself is negligible, I might just put it back to how I originally had it, since it makes the code a bit clearer.